azure_functions_worker/bindings/shared_memory_data_transfer/file_accessor_factory.py

```diff
@@ -13,6 +18,9 @@ class FileAccessorFactory:
     """
     @staticmethod
     def create_file_accessor():
-        if os.name == 'nt':
+        if sys.platform == "darwin" and not is_envvar_true(
```

Comment:
Disabling the whole SharedMemory feature for Mac if the feature is not on by default.